### PR TITLE
RL_on_GCP (follow up)

### PR DESCRIPTION
**Pull request by lerrytang**
You should be able to run the code now. E.g.,
```
bash start.sh -p google-project-id -a TD3
```

Major changes are:
1. add run instructions and merits of using CMLE in README, removed img.
2. add agents implementations (DDPG, TD3, C2A2).
3. move blog related code into sub folder.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/cloudml-samples/329)

### CODE REVIEW

File: reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+"""Implementation of a TD3 agent.
+
+Implementation of TD3 - Twin Delayed Deep Deterministic Policy Gradient
+Algorithm and hyperparameter details can be found here:
+https://arxiv.org/pdf/1802.09477.pdf
+
+"""
+import agent
+from common import replay_buffer
+from common.actor_critic import ActorNetwork, CriticNetwork
```
Comment (gogasca) [inline reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py]: nit: Each import should be in one line:

```
from common.actor_critic import ActorNetwork
from common.actor_critic import CriticNetwork
```

File: reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import numpy as np
+
+
+class TD3(agent.Agent):
+    """TD3 agent."""
+
+    def __init__(self, env, sess, config):
+        """Initialize members."""
+        self.env = env
```
Comment (gogasca) [inline reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py]: nit: Move self.env one line below to separate: Class variables and method variables.

File: reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        state_dim = env.observation_space.shape[0]
+        self.action_dim = env.action_space.shape[0]
+        self.action_high = env.action_space.high
+        self.action_low = env.action_space.low
+        self.batch_size = config.batch_size
+        self.warmup_size = config.warmup_size
+        self.gamma = config.gamma
+        self.sigma = config.sigma
+        self.sigma_tilda = config.sigma_tilda
+        self.c = config.c
```
Comment (gogasca) [inline reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py]: nit: rename self.c to something more relevant for the non-RL expert.

File: reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        self.d = config.d
```
Comment (gogasca) [inline reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py]: Same as above

File: reinforcement_learning/rl_on_gcp_demo/trainer/ddpg_agent.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+"""Implementation of a DDPG agent.
+
+Implementation of DDPG - Deep Deterministic Policy Gradient
+Algorithm and hyperparameter details can be found here:
+http://arxiv.org/pdf/1509.02971v2.pdf
+
+"""
+import agent
+from common import replay_buffer
+from common.actor_critic import ActorNetwork, CriticNetwork
```
Comment (gogasca) [inline reinforcement_learning/rl_on_gcp_demo/trainer/ddpg_agent.py]: nit: Each import into a single line:

```
from common.actor_critic import ActorNetwork
from common.actor_critic import CriticNetwork
```

File: reinforcement_learning/rl_on_gcp_demo/trainer/ddpg_agent.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+import numpy as np
+
+
+class DDPG(agent.Agent):
+    """DDPG agent."""
+
+    def __init__(self, env, sess, config):
+        """Initialize members."""
+        self.env = env
```
Comment (gogasca) [inline reinforcement_learning/rl_on_gcp_demo/trainer/ddpg_agent.py]: nit: Move self.env one line below to separate: Class variables and method variables.

File: reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py (hunk 1 above)
Comment (lerrytang) [reply reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py]: fixed in ac35cd106aa30d9f4e0cc69898e7da768c8518a7

File: reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py (hunk 2 above)
Comment (lerrytang) [reply reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py]: fixed in ac35cd106aa30d9f4e0cc69898e7da768c8518a7

File: reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py (hunk 3 above)
Comment (lerrytang) [reply reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py]: fixed in ac35cd106aa30d9f4e0cc69898e7da768c8518a7

File: reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py (hunk 4 above)
Comment (lerrytang) [reply reinforcement_learning/rl_on_gcp_demo/trainer/td3_agent.py]: fixed in ac35cd106aa30d9f4e0cc69898e7da768c8518a7

File: reinforcement_learning/rl_on_gcp_demo/trainer/ddpg_agent.py (hunk 5 above)
Comment (lerrytang) [reply reinforcement_learning/rl_on_gcp_demo/trainer/ddpg_agent.py]: fixed in ac35cd106aa30d9f4e0cc69898e7da768c8518a7

File: reinforcement_learning/rl_on_gcp_demo/trainer/ddpg_agent.py (hunk 6 above)
Comment (lerrytang) [reply reinforcement_learning/rl_on_gcp_demo/trainer/ddpg_agent.py]: fixed in ac35cd106aa30d9f4e0cc69898e7da768c8518a7

File: reinforcement_learning/rl_on_gcp_demo/trainer/ddpg_agent.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+"""
+import agent
+from common import replay_buffer
+from common.actor_critic import ActorNetwork
+from common.actor_critic import CriticNetwork
+import numpy as np
+
+
+class DDPG(agent.Agent):
```
Comment (dizcology) [inline reinforcement_learning/rl_on_gcp_demo/trainer/ddpg_agent.py]: How does the proposed `agent` module compare with the existing `tf-agents`?  https://github.com/tensorflow/agents/blob/master/tf_agents/agents/tf_agent.py

(`tf-agents` comes with also TD3 and DDPG implementations, and the other parts of the library might conform better to interfaces by deepmind and openai-gym.)

Comment (dizcology) [reply reinforcement_learning/rl_on_gcp_demo/trainer/ddpg_agent.py]: please at least point out and comment on the existing tf-agents library in both the blog post and in the README of this set of samples if you don't have time to change the code or find a way to consolidate the different interfaces (your `agent` and `tf-agents`).

**Pull request merged**